main.go

```diff
@@ -117,19 +120,25 @@ func getDefaultGatewayIfaceName() (string, error) {
 			ifaceName = route.Iface.Name
 		}
 
-		isDefault := "false"
-		if route.Default {
-			isDefault = "true"
-		}
-
-		verboseLog(fmt.Sprintf("Route %d: IFace = %s, Default = %s", i, ifaceName, isDefault))
+		verboseLog(fmt.Sprintf("Route %d: IFace = %s, Default = %t", i, ifaceName, route.Default))
 		if route.Default {
 			if route.Iface == nil {
 				return "", errors.New("found default route but could not determine interface")
 			}
-			return route.Iface.Name, nil
+			defaultInterfaces = append(defaultInterfaces, route.Iface.Name)
 		}
 	}
+
+	if len(defaultInterfaces) > 0 {
+		// sort interfaces  so we pick the first one properly
+		sort.Slice(defaultInterfaces, func(i, j int) bool {
+			return defaultInterfaces[i] < defaultInterfaces[j]
```

Comment:
I think `10.100.0.1` is considered smaller than `10.2.0.1` when you do it like that. Should that be unit tested?

Reply:
you are talking a IPs,  I am talking interfaces so its eth2 vs eth0 (or bond1 vs bond0), but sure i can unit test it

Reply:
That's ok then. Unless we have more that 10.

Reply:
But is should always start with 0 so all good.